pkg/client/metadata.go

```diff
@@ -0,0 +1,135 @@
+package client
+
+import (
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/buildpacks/lifecycle/platform"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+type TagInfo struct {
+	Name    string
+	Message string
+	Type    string
+	TagHash string
+	TagTime time.Time
+}
+
+func GitMetadata(appPath string) *platform.ProjectSource {
+	// appPath2 := "/Users/nitish/OSS/pack"
+	repo, err := git.PlainOpen(appPath)
+	if err != nil {
+		print("unable to open git repo")
```

Comment:
Ultimately, I would remove the `print` statements here, and just return `nil` if there is an issue in these cases